main/src/hadd.cxx

```diff
@@ -1,39 +1,47 @@
 /**
   \file hadd.cxx
-  \brief This program will add histograms (see note) and Trees from a list of root files and write them to a target root file.
-  The target file is newly created and must not be
-  identical to one of the source files.
+  \brief This program will add histograms (see note) and Trees from a list of root files and write them to a target root
+  file. The target file is newly created and must not be identical to one of the source files.
 
   Syntax:
   ```{.cpp}
-       hadd targetfile source1 source2 ...
+       hadd [flags] targetfile source1 source2 ...
   ```
-  or
-  ```{.cpp}
-       hadd -f targetfile source1 source2 ...
-  ```
-  (targetfile is overwritten if it exists)
+
+  Flags can be passed before or after the positional arguments.
+  The first positional (non-flag) argument will be interpreted as the targetfile.
+  After that, the first sequence of positional arguments will be interpreted as the input files.
+  If two sequences of positional arguments are separated by flags, all sequences but the first
+  will be ignored.
+  If a flag requires an argument, the argument can be specified in any of these ways:
+
+     # All equally valid:
+     -j 16
+     -j16
+     -j=16
+
+   The first syntax is the preferred one since it's backward-compatible with previous versions of hadd.
+   The -f flag is an exception to this rule: it only supports the `-f[0-9]` syntax.
+
+   Note that merging multiple flags is NOT supported: `-jfa` will be interpreted as -j=fa, which is invalid!
```

Comment:
Nice, there should be a test to double check that this situation raises an error

Reply:
Did we add this test?

Reply:
Yes: https://github.com/root-project/roottest/pull/1201/files#diff-031cbdf502ed731ba47ea11e7ee4fce1b12af001774c310bc2e5f46c988e1ab8R60-R64